### PR TITLE
Take into account slice thickness in RetrieveSOPInstanceUIDFromZPosition

### PR DESCRIPTION
**Pull request by tbaudier**
**Bug:**
I have RTStruct slices located in [-102.134, -99.1341, -96.1341] along z-axis and the CT slices are located in [-102, -99, -96]. So the SOPInstanceUID was empty because the difference between both was >0.01.

**Solution:**
I propose to take into account the slice thickness (tag 0x0018, 0x0050) instead of the value of 0.01 if this tag is present.

### CODE REVIEW

**Comment by malaterre** [comment]
@tbaudier Could please confirm your code intent to use `SliceThickness` and not `SpacingBetweenSlices`

**Comment by tbaudier** [comment]
You are right. I change to ```SpacingBetweenSlices``` to avoid overlap problem. I just use the ```fabs``` to deal negative value.

**Comment by malaterre** [comment]
@tbaudier I fail to understand the `/2.0`. The definition I see:


> Spacing Between Slices | (0018,0088) | 3 | Spacing between slices, in mm. The spacing is measured from the center-to-center of each slice.


ref:
* http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.8.3.html#sect_C.8.3.1

**Comment by tbaudier** [comment]
Hi @malaterre 

sbs

The code looked for the correct CT slice for the RTStruct according z-position. So I tried to find the closest CT slice. This is why I divided by 2.0. Because, if 

```math
z_RTStruct - z_N > SpacingBetweenSlice/2.0
```
Then the RTstuct is closer to the N+1 CT slice

File: Source/MediaStorageAndFileFormat/gdcmDirectoryHelper.cxx [hunk 1]
```diff
@@ -174,6 +174,15 @@ std::string DirectoryHelper::RetrieveSOPInstanceUIDFromZPosition(double inZPos,
...
   Tag thePosition(0x0020, 0x0032);
   Tag theSOPInstanceUID(0x0008, 0x0018);
+  Tag theSpacingBetweenSlice(0x0018, 0x0088);
+  double interSlice = 0.01;
+  if (inDS.begin() != inDS.end() && inDS.begin()->FindDataElement(theSpacingBetweenSlice))
+    {
+    DataElement tmpDe = inDS.begin()->GetDataElement(theSpacingBetweenSlice);
+    Attribute<0x0018,0x0088> tmpAt;
+    tmpAt.SetFromDataElement(tmpDe);
+    interSlice = fabs(tmpAt.GetValue())/2.0;
```
Comment (malaterre) [inline Source/MediaStorageAndFileFormat/gdcmDirectoryHelper.cxx]: Remove `fabs` here

**Comment by malaterre** [comment]
@tbaudier OK that makes sense now, please remove the extra `fabs` and I'll merge.

**Comment by malaterre** [comment]
> I have an example with such a case

From real vendor (siemens/philips/gems) ?

**Comment by tbaudier** [comment]
A CT image from a PHILIPS-705 Brilliance Big Bore machine

**Comment by malaterre** [comment]
OK merged ! Thanks for the contribution ! I still fail to understand what it means a negative spacing in this case. 

Just for reference, here is what the standard states for Enhanced CT Image Storage:

* http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.16.2.html#sect_C.7.6.16.2.1

> Spacing between adjacent slices, in mm. The spacing is measured from the center-to-center of each slice, and if present shall not be negative.

**Comment by tbaudier** [comment]
Thank you for the merging. Negative spacing is also a conundrum for me!

**Comment by divyarathore** [comment]
Hi all,

Can someone please point out to a good online resource about DCE-MRI (aka,
Enhanced MRI, perfusion MRI) DICOM file structure, please?

Additionally, a sample gdcm example would be great.

I am just starting on GDCM and wanted to integrate it in a post processing
tool.

Best regards,
D

**Pull request merged**